nixos/modules/programs/opengamepadui.nix

```diff
@@ -0,0 +1,295 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.opengamepadui;
+  gamescopeCfg = config.programs.gamescope;
+
+  opengamepadui-gamescope =
+    let
+      exports = builtins.attrValues (
+        builtins.mapAttrs (n: v: "export ${n}=${v}") cfg.gamescopeSession.env
+      );
+    in
+    # Based on gamescope-session-plus from ChimeraOS
+    pkgs.writeShellScriptBin "opengamepadui-gamescope" ''
+      ${builtins.concatStringsSep "\n" exports}
+
+      # Enable Mangoapp
+      export MANGOHUD_CONFIGFILE=$(mktemp /tmp/mangohud.XXXXXXXX)
+      export RADV_FORCE_VRS_CONFIG_FILE=$(mktemp /tmp/radv_vrs.XXXXXXXX)
+
+      # Plop GAMESCOPE_MODE_SAVE_FILE into $XDG_CONFIG_HOME (defaults to ~/.config).
+      export GAMESCOPE_MODE_SAVE_FILE="''${XDG_CONFIG_HOME:-$HOME/.config}/gamescope/modes.cfg"
+      export GAMESCOPE_PATCHED_EDID_FILE="''${XDG_CONFIG_HOME:-$HOME/.config}/gamescope/edid.bin"
+
+      # Make path to gamescope mode save file.
+      mkdir -p "$(dirname "$GAMESCOPE_MODE_SAVE_FILE")"
+      touch "$GAMESCOPE_MODE_SAVE_FILE"
+
+      # Make path to Gamescope edid patched file.
+      mkdir -p "$(dirname "$GAMESCOPE_PATCHED_EDID_FILE")"
+      touch "$GAMESCOPE_PATCHED_EDID_FILE"
+
+      # Initially write no_display to our config file
+      # so we don't get mangoapp showing up before OpenGamepadUI initializes
+      # on OOBE and stuff.
+      mkdir -p "$(dirname "$MANGOHUD_CONFIGFILE")"
+      echo "no_display" >"$MANGOHUD_CONFIGFILE"
+
+      # Prepare our initial VRS config file
+      # for dynamic VRS in Mesa.
+      mkdir -p "$(dirname "$RADV_FORCE_VRS_CONFIG_FILE")"
+      echo "1x1" >"$RADV_FORCE_VRS_CONFIG_FILE"
+
+      # To play nice with the short term callback-based limiter for now
+      export GAMESCOPE_LIMITER_FILE=$(mktemp /tmp/gamescope-limiter.XXXXXXXX)
+
+      ulimit -n 524288
+
+      # Setup socket for gamescope
+      # Create run directory file for startup and stats sockets
+      tmpdir="$([[ -n ''${XDG_RUNTIME_DIR+x} ]] && mktemp -p "$XDG_RUNTIME_DIR" -d -t gamescope.XXXXXXX)"
+      socket="''${tmpdir:+$tmpdir/startup.socket}"
+      stats="''${tmpdir:+$tmpdir/stats.pipe}"
+
+      # Fail early if we don't have a proper runtime directory setup
+      if [[ -z $tmpdir || -z ''${XDG_RUNTIME_DIR+x} ]]; then
+      	echo >&2 "!! Failed to find run directory in which to create stats session sockets (is \$XDG_RUNTIME_DIR set?)"
+      	exit 0
+      fi
+
+      export GAMESCOPE_STATS="$stats"
+      mkfifo -- "$stats"
+      mkfifo -- "$socket"
+
+      # Start gamescope compositor, log it's output and background it
+      echo gamescope ${builtins.toString cfg.gamescopeSession.args}  -R $socket -T $stats >"$HOME"/.gamescope-cmd.log
```

Comment:
I'm not sure if using `toString` is wise here. `lib.escapeShellArgs` would be better.